### PR TITLE
Refactor aws client

### PR DESCRIPTION
**Pull request by janboll**
Added an interface Client, that abstracts AWS client calls. This indirection helps using gomock to create tests. See ec2_test.go as an example for this. This is a pattern proposed by AWS to mock the AWS client calls and enable writing tests. See: https://aws.amazon.com/blogs/developer/mocking-out-then-aws-sdk-for-go-for-unit-testing/

As of this commit only the client usage of ec2.go and parts of route53.go have been updated. The rest of the codebase still requires this refactoring.

### CODE REVIEW

File: pkg/route53.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func (e *Route53Exporter) getHostedZonesPerAccountMetrics(client *servicequotas.
 // CollectLoop runs indefinitely to collect the route53 metrics in a cache. Metrics are only written into the cache once all have been collected to ensure that we don't have a partial collect.
 func (e *Route53Exporter) CollectLoop() {
 	route53Svc := route53.New(e.sess)
-	serviceQuotaSvc := servicequotas.New(e.sess)
+	awsclient := awsclient.NewClientFromSession(e.sess)
```
Comment (patjlm) [inline pkg/route53.go:115]: I would not name the variable the same than the module being imported. Maybe just `client` ?

File: pkg/awsclient/awsclient.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	ListRequestedServiceQuotaChangeHistoryByQuota(*servicequotas.ListRequestedServiceQuotaChangeHistoryByQuotaInput) (*servicequotas.ListRequestedServiceQuotaChangeHistoryByQuotaOutput, error)
+}
+
+type awsClient struct {
+	ec2Client           ec2iface.EC2API
+	serviceQuotasClient servicequotasiface.ServiceQuotasAPI
+}
+
+// NewAwsClientInput input for new aws client
+type NewAwsClientInput struct {
```
Comment (patjlm) [inline pkg/awsclient/awsclient.go:37]: where is this used ?

Comment (janboll) [reply pkg/awsclient/awsclient.go:37]: :broom:

**Review by patjlm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**